pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -360,6 +360,92 @@ func TestPriorityQueue_Delete(t *testing.T) {
 	}
 }
 
+func BenchmarkMoveAllToActiveOrBackoffQueue(b *testing.B) {
+	tests := []struct {
+		name                 string
+		podsInUnschedulableQ int
+		moveEvent            string
+	}{
+		{
+			name:      "baseline",
+			moveEvent: UnschedulableTimeout,
+		},
+		{
+			name:      "worst",
+			moveEvent: NodeAdd,
+		},
+		{
+			name: "random",
+			// leave "moveEvent" unspecified
+		},
+	}
+
+	podTemplates := []*v1.Pod{
+		highPriorityPodInfo.Pod, highPriNominatedPodInfo.Pod,
+		medPriorityPodInfo.Pod, unschedulablePodInfo.Pod,
+	}
+	events := []string{NodeAdd, NodeTaintChange, NodeAllocatableChange, NodeConditionChange, NodeLabelChange, PodAdd,
+		PvcAdd, PvcUpdate, PvAdd, PvUpdate, StorageClassAdd, CSINodeAdd, CSINodeUpdate}
+	pluginNum := 20
+	var plugins []string
+	// Mimic that we have 20 plugins loaded in runtime.
+	for i := 0; i < pluginNum; i++ {
+		plugins = append(plugins, fmt.Sprintf("fake-plugin-%v", i))
+	}
+
+	for _, tt := range tests {
+		for _, podsInUnschedulableQ := range []int{1000, 5000} {
+			b.Run(fmt.Sprintf("%v-%v", tt.name, podsInUnschedulableQ), func(b *testing.B) {
+				for i := 0; i < b.N; i++ {
+					b.StopTimer()
+					c := clock.NewFakeClock(time.Now())
+
+					m := make(map[framework.ClusterEvent]sets.String)
+					if tt.moveEvent != "" {
+						// Simulate that all plugins register the NodeAdd event.
```

Comment:
why? I think we should always distribute events over plugins like the logic below.

Reply:
We could, but that's a bit overkill. I updated the code to ensure that the `m` here always holds `len(events)` items, instead of 1, which will be good enough to simulate a practical execution path.

Reply:
iiuc, the reason to just register all plugins into NodeAdd is to have an extreme case. But you can do that with the distribution you have below. So getting rid of this conditional leaves the same behavior while being more readable.

Reply:
SG, agree it can make the code more readable.